float32 timeinmillis
float32 voltage
float32 temperature